lib/kino/smart_cell/chart_builder.ex

```diff
@@ -74,9 +72,20 @@ defmodule Kino.SmartCell.ChartBuilder do
   @impl true
   def handle_info({:scan_binding_result, data_options, vl_alias}, ctx) do
     ctx = assign(ctx, data_options: data_options, vl_alias: vl_alias)
-    broadcast_event(ctx, "set_available_data", %{"data_options" => data_options})
 
-    {:noreply, ctx}
+    if !ctx.assigns.fields["data_variable"] do
+      data_variable = Map.keys(data_options) |> List.first()
+      updated_fields = updates_for_data_variable(ctx, data_variable)
+      ctx = update(ctx, :fields, &Map.merge(&1, updated_fields))
+      broadcast_event(ctx, "update", %{"fields" => updated_fields})
+      broadcast_event(ctx, "set_available_data", %{"data_options" => data_options})
+
+      {:noreply, ctx}
+    else
+      broadcast_event(ctx, "set_available_data", %{"data_options" => data_options})
+
+      {:noreply, ctx}
+    end
```

Comment:
We can broadcast just once, like this:

```suggestion
    updated_fields =
      case {ctx.assigns.fields["data_variable"], data_options} do
        {nil, [{data_variable, _} | _]} -> updates_for_data_variable(ctx, data_variable)
        _ -> %{}
      end

    ctx = update(ctx, :fields, &Map.merge(&1, updated_fields))

    broadcast_event(ctx, "set_available_data", %{
      "data_options" => data_options,
      "fields" => updated_fields
    })
```

And on the JS side we just need to do `setValues(fields);` in the handler, as would happen with the separate event.

Note: the code above assumes that `data_options` is a list of pairs, instead of a map, and we should make that change. Just recently we changed Livebook so that variables are ordered from latest to oldest, so we need a list to keep the order :)